todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/taskdetail/TaskDetailFragment.java

```diff
@@ -68,13 +89,60 @@ public static TaskDetailFragment newInstance(String taskId) {
     @Override
     public void onResume() {
         super.onResume();
-        mPresenter.subscribe();
+        bind();
     }
 
     @Override
     public void onPause() {
+        unbind();
         super.onPause();
-        mPresenter.unsubscribe();
+    }
+
+    private void bind() {
+        mSubscription = new CompositeSubscription();
```

Comment:
here and in the other views you just flipping the bindings from the P to the V. 
Essentially you coupling V with binding declaration, binding implementation and lots of presenter. V becomes way too complex.